distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/cli/ShowKeyStoreCommand.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings.cli;
+
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.settings.BaseKeyStoreCommand;
+import org.elasticsearch.common.settings.KeyStoreWrapper;
+import org.elasticsearch.core.PathUtils;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.core.internal.io.Streams;
+import org.elasticsearch.env.Environment;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A subcommand for the keystore cli to show the value of a setting in the keystore.
+ */
+public class ShowKeyStoreCommand extends BaseKeyStoreCommand {
+
+    private final OptionSpec<String> arguments;
+    private final OptionSpec<String> outputOption;
+
+    public ShowKeyStoreCommand() {
```

Comment:
We used to have that flag and explicitly removed it in #41701
The issue at the time was forcing orchestration tools to know when to add a string vs a file.
  
When settings are _used_ the server code know whether it wants a String or byte[] (InputStream) because it knows what that setting means. It doesn't really care how it was added - it's an error to add binary data that's not valid utf-8, but _read from a file_ doesn't imply _isn't pure utf-8 text_.

The issue here is the same. If it is convenient for me to do:
```
gen-password > bootstrap_password.txt
elasticsearch-keystore add-file bootstrap.password bootstrap_password.txt
```
The choice of `add-file` shouldn't prevent me from showing the password on the terminal.

Reply:
I had forgotten about 41701. I can't comprehend now what the pain for operators was around this but I don't challenge it was a real one. Since we don't want to differentiate between file and string on input, my comment is irrelevant

Reply:
It was ECK specifically.
Prior to this, ECK needed to know whether to call `add` or `add-file` for each secure setting that was configured as a k8s secret. Which means it needed to know about every ES secure setting.
That wasn't good